tests/core/pthread/test_pthread_exit.c

```diff
@@ -0,0 +1,45 @@
+#include <assert.h>
+#include <pthread.h>
+#include <stdlib.h>
+#include <stdio.h>
+
+/* This test aims to check the following assertion:
+ *
+ * pthread_exit() terminates the current thread and
+ * allows a call to pthread_join() on this thread retrieve
+ * the argument (if the thread is not detached).
+ *
+ * We do this twice to ensure that the thread has exited
+ * gracefully and that we are able to recreate the thread.
```

Comment:
I would have thought we have tests that cover these basics already don't we?   I guess this is specific to pthread_exit rather then returning from the thread entry point?

Reply:
I thought it would be useful to include a simplified test to make this issue easier to reproduce on the master branch. But, indeed this should already be covered with `test_pthread_exit_1_2` in posixtestsuite which is enabled within commit 38b4bde9a81ad67951846efc7177ba82dab12ef0.

Removed this test with commit d0f3bc0d316abf34afca1c95f65baec91d22179d.

> I guess this is specific to pthread_exit rather then returning from the thread entry point?

Yes, I think this issue only manifests  when a call to `pthread_exit` is made explicitly, returning from the thread entry point should work without problems.